common/script.c

```diff
@@ -0,0 +1,325 @@
+/*
+ * Phoenix-RTOS
+ *
+ * script parsing
+ *
+ * Copyright 2020 Phoenix Systems
+ *
+ * Author: Gerard Swiderski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include "script.h"
+
+#define LOG_ERROR(...)	fprintf(stderr, __VA_ARGS__)
+
+
+#define IS_SPACE(c)   ((c) == ' '  || (c) == '\t')
+#define IS_NEWLINE(c) ((c) == '\n' || (c) == '\r')
+#define IS_DIGIT(c)   ((c) >= '0'  && (c) <= '9' )
+#define IS_QUOTE(c)   ((c) == '\"' || (c) == '\'')
+
+
+/* initialize parser, and load psu script file */
+int script_load(script_t *s, const char *fname)
+{
+	int fd;
+	void *ptr = MAP_FAILED;
+	struct stat statbuf = {0};
+
+	if ((fd = open(fname, O_RDONLY)) < 0) {
+		LOG_ERROR("Unable to open '%s' script file.\n", fname);
+		return SCRIPT_ERROR;
+	}
+
+	if (fstat(fd, &statbuf) < 0) {
+		LOG_ERROR("fstat() failed.\n");
+		goto on_error;
+	}
+
+	if (statbuf.st_size > 1024*100) {
+		LOG_ERROR("Script file too big (100kB limit).\n");
+		goto on_error;
+	}
+
+	ptr = mmap(NULL, statbuf.st_size, PROT_READ, MAP_PRIVATE, fd, 0);
+
+	if (ptr == MAP_FAILED) {
+		LOG_ERROR("Unable to allocate memory.\n");
+		goto on_error;
```

Comment:
It would be even easier to just close the file here, and don't check for MAP_FAILED later. All to remove this goto ;)